### PR TITLE
Add hyperlink recommended rules to the appendix

### LINKED ISSUE #76: Feature request: hyperlinks on text
Hi,

Hyperlinks are convenient in bibliographies to access the paper directly. However including the url as a text uses space, and is against what hyperlinks are all about. I would like to be able to add a hyperlink to the url field in the bibtex entry to the title for example. As far as I understood it is not currently possible with CSL.

Is it possible to add this feature in the future? Either add a url property to text, or add a sub-item url.

Thanks,

### PR DESCRIPTION
**Pull request by bdarcus**
Adds a section of the specification that suggests how processors 
and applications should output links on references.

addresses #76

### CODE REVIEW

**Comment by bdarcus** [comment]
Note: I wasn't sure where to put this, so added an appendix section.

File: specification.rst [hunk 1]
```diff
@@ -2854,3 +2854,21 @@ The page abbreviation rules for the different values of the
...
+
+Appendix VI: Links
+------------------
+
+The CSL syntax does not have support for configuration of links. However,
+processors should include links on bibliographic references, using the following
+rules:
+
+If the item includes the following identifiers, the target of the link should be
+set, in order of priority:
```
Comment (bdarcus) [inline specification.rst:2867]: This assumes only one URL is printed in the output. I suppose if we want to retain this restriction, we could make that explicit in this text.

File: specification.rst [hunk 2]
```diff
@@ -2854,3 +2854,21 @@ The page abbreviation rules for the different values of the
...
+
+1. ``url``: output as is
+2. ``doi``: prepend with ``https://dx.doi.org/``
```
Comment (bwiernik) [inline specification.rst:2870]: Should be `prepend with ``https://doi.org/```

File: specification.rst [hunk 3]
```diff
@@ -2854,3 +2854,21 @@ The page abbreviation rules for the different values of the
...
+3. ``pmid``: prepend with ``https://www.ncbi.nlm.nih.gov/pubmed/``
```
Comment (bwiernik) [inline specification.rst:2871]: Should also include `pmcid` in this list.

File: specification.rst (hunk 2 above)
Comment (bwiernik) [reply specification.rst:2870]: I'm unsure of the priority that should be given to `doi`/`pmid`/`pmcid` vs. `url`. The first three are intended to be stable, unchanging identifiers that are more reliable than URLs, but some publishers (namely Taylor and Francis) are pretty bad at registering their DOIs and ensuring that they resolve in a timely manner.

Comment (bdarcus) [reply specification.rst:2870]: I was unsure also. 

I also considered whether to create separate lists.

Comment (bdarcus) [reply specification.rst:2870]: But I was hoping we could come up with some solid suggestions worth putting in the spec, but also leave room for iteration over time.

File: specification.rst (hunk 1 above)
Comment (bwiernik) [reply specification.rst:2867]: We could say, "If there is only one identifier, link the whole reference. If there are identifiers, link them individually."

File: specification.rst (hunk 2 above)
Comment (bwiernik) [reply specification.rst]: @adam3smith What do you think regarding DOI vs URL prioirity?

**Comment by denismaier** [comment]
What if I don't want any links? Is there a way to specify this?

File: specification.rst (hunk 2 above)
Comment (denismaier) [reply specification.rst]: I'd say stable identifiers first.

**Comment by bdarcus** [comment]
> What if I don't want any links? Is there a way to specify this?

I think that should be selected on the application; as a user preference, global and/or document-specific.

**Comment by bwiernik** [comment]
We can instruct processors to have a flag to enable/disable it.

**Comment by bwiernik** [comment]
@bdarcus I revised to address the issues that were raised here. What do you think?

For comparison, Zotero's current behavior with citeproc-js is to link an identifier if it is rendered, but not generate a link at all if no link is rendered. That might be a better default behavior than anchoring a link to `title` if no identifier is rendered, even though that goes against CrossRef recommendations. With that default, the option flag could enable link-setting when no identifier is rendered. (I don't really see a reason to provide for disabling links on the identifiers when they are rendered.)

**Comment by bdarcus** [comment]
> @bdarcus I revised to address the issues that were raised here. What do you think?

Looks good; thanks!

It's just a recommendation, so we can see how it goes and adjust if needed.

**Pull request merged**